src/base/PolymeshTransactionBase.ts

```diff
@@ -468,4 +590,103 @@ export abstract class PolymeshTransactionBase<Values extends unknown[] = unknown
   ): void {
     resolve(receipt);
   }
+
+  /**
+   * @hidden
+   *
+   * Check if balances and allowances (both third party and signing Account)
+   *   are sufficient to cover this transaction's fees
+   */
+  private async assertFeesCovered(): Promise<void> {
+    const {
+      fees: { total },
+      payingAccountData,
+    } = await this.getFees();
+
+    const { type, balance } = payingAccountData;
+
+    if (type === PayingAccountType.Subsidy) {
+      const { allowance } = payingAccountData;
+      if (!this.supportsSubsidy()) {
+        throw new PolymeshError({
+          code: ErrorCode.UnmetPrerequisite,
+          message: 'This transaction cannot be run by a subsidized Account',
+        });
+      }
+
+      if (allowance.lt(total)) {
+        throw new PolymeshError({
+          code: ErrorCode.UnmetPrerequisite,
+          message:
+            "The subsidizer Account has not granted the caller Account enough allowance to pay this transaction's fees",
```

Comment:
```suggestion
            "Insufficient subsidy allowance to pay this transaction's fees",
```
(nit) I think this can be less verbose